astropy/io/fits/fitsrec.py

```diff
@@ -497,6 +497,7 @@ def __getitem__(self, key):
 
         # We got a view; change it back to our class, and add stuff
         out = out.view(type(self))
+        out._uint = self._uint
```

Comment:
I vaguely recall adding some of the above code, but looking at it again I wonder a bit why... Normally, an `ndarray` subclass is supposed to do this in `__array_finalize__`, but here of course the view as a `recarray` and then the view back here circumvent that (and from a quick look, just doing `out.__arrray_finalize__(self)` would seem unlikely to do quite the right thing)

Anyway, I think the fix here is good, though perhaps have a quick look at `__array_finalize__` and see if anything else is missing (since `_uint` at least is updated there).

Reply:
Yes, I should have mentioned that I requested your review because you changed a bit this code in the past (e.g. #6955), but the issue was already there before. The problem is that with the recarray view we loose the FITS_rec attributes, so they are no more available in `__array_finalize__` when viewing back to FITS_rec.
I'm not completely sure about the other attributes, I think that the missing ones can be recomputed from the coldefs.